pennylane/transforms/adjoint_metric_tensor.py

```diff
@@ -157,56 +135,59 @@ def circuit(weights):
     shot simulations.
     """
     if isinstance(circuit, qml.tape.QuantumScript):
-        return _adjoint_metric_tensor_tape(circuit, device)
+        return _adjoint_metric_tensor_tape(circuit)
     if isinstance(circuit, (qml.QNode, qml.ExpvalCost)):
         return _adjoint_metric_tensor_qnode(circuit, device, hybrid)
 
     raise qml.QuantumFunctionError("The passed object is not a QuantumTape or QNode.")
 
 
-def _adjoint_metric_tensor_tape(tape, device):
+def _adjoint_metric_tensor_tape(tape):
     """Computes the metric tensor of a tape using the adjoint method and a given device."""
     # pylint: disable=protected-access
-    if device.shots is not None:
+    if tape.shots:
         raise ValueError(
             "The adjoint method for the metric tensor is only implemented for shots=None"
         )
+    if set(tape.wires) != set(range(tape.num_wires)):
+        wire_map = {w: i for i, w in enumerate(tape.wires)}
+        tape = qml.map_wires(tape, wire_map)
     tape = qml.transforms.expand_trainable_multipar(tape)
 
     # Divide all operations of a tape into trainable operations and blocks
     # of untrainable operations after each trainable one.
     trainable_operations, group_after_trainable_op = _group_operations(tape)
 
-    dim = 2**device.num_wires
+    dim = 2**tape.num_wires
     # generate and extract initial state
-    psi = device._create_basis_state(0)
+    prep = tape[0] if len(tape) > 0 and isinstance(tape[0], qml.operation.StatePrep) else None
```

Comment:
just commenting so it doesn't cause a conflict - Jay's change might beat this one, and if it doesn't someone should tell him :)

Reply:
@Jaybsoni

Reply:
thansk for catching that.